torchgen/dest/lazy_ir.py

```diff
@@ -307,11 +308,22 @@ def metrics(self, func: NativeFunction, schema: LazyIrSchema) -> str:
 
     def get_device(self, func: NativeFunction, schema: LazyIrSchema) -> str:
         value_args = schema.filtered_args(values=True, scalars=False)
+        scalar_args = schema.filtered_args(values=False, scalars=True)
         value_types_names = [f"{a.name}" for a in value_args if not a.is_wrapped_scalar]
-        assert (
-            len(value_types_names) > 0
-        ), "Code below assumes there is at least one tensor arg"
-        return f"""auto common_device = torch::lazy::GetBackendDevice({', '.join(value_types_names)});
+        if len(value_types_names) > 0:
+            get_device_str = (
+                f"torch::lazy::GetBackendDevice({', '.join(value_types_names)})"
+            )
+        else:
+            optional_device = OptionalCType(BaseCType(deviceT))
+            for a in scalar_args:
+                if a.lazy_type == optional_device:
+                    get_device_str = f"atenDeviceToBackendDevice({a.name})"
```

Comment:
is this proposed behavior actually correct?  it looks like it would only try the first arg that has an optional device, but if there are 2 args with optional devices and only the second one is non-nullopt, it will return nullopt